src/subcommand/wallet/burn.rs

```diff
@@ -28,68 +28,88 @@ pub struct Burn {
-      format!(
-        "metadata length {} over maximum {}",
-        metadata.len(),
-        u32::MAX
-      )
-    })?;
-    builder = builder.push_slice(push);
-
-    // pad OP_RETURN script to least five bytes to ensure transaction base size
-    // is greater than 64 bytes
-    let padding = 5usize.saturating_sub(builder.as_script().len());
-    if padding > 0 {
-      // subtract one byte push opcode from padding length
-      let padding = vec![0; padding - 1];
-      let push: &script::PushBytes = padding.as_slice().try_into().unwrap();
-      builder = builder.push_slice(push);
-    }
+        let push: &script::PushBytes = metadata.as_slice().try_into().with_context(|| {
+          format!(
+            "metadata length {} over maximum {}",
+            metadata.len(),
+            u32::MAX
+          )
+        })?;
+        builder = builder.push_slice(push);
 
-    let script_pubkey = builder.into_script();
+        // pad OP_RETURN script to least five bytes to ensure transaction base size
+        // is greater than 64 bytes
+        let padding = 5usize.saturating_sub(builder.as_script().len());
+        if padding > 0 {
+          // subtract one byte push opcode from padding length
+          let padding = vec![0; padding - 1];
+          let push: &script::PushBytes = padding.as_slice().try_into().unwrap();
+          builder = builder.push_slice(push);
+        }
 
-    ensure!(
-      self.no_limit || script_pubkey.len() <= MAX_STANDARD_OP_RETURN_SIZE,
-      "OP_RETURN with metadata larger than maximum: {} > {}",
-      script_pubkey.len(),
-      MAX_STANDARD_OP_RETURN_SIZE,
-    );
+        let script_pubkey = builder.into_script();
 
-    let burn_amount = Amount::from_sat(1);
+        ensure!(
+          self.no_limit || script_pubkey.len() <= MAX_STANDARD_OP_RETURN_SIZE,
+          "OP_RETURN with metadata larger than maximum: {} > {}",
+          script_pubkey.len(),
+          MAX_STANDARD_OP_RETURN_SIZE,
+        );
 
-    let unsigned_transaction = Self::create_unsigned_burn_transaction(
-      &wallet,
-      inscription_info.satpoint,
-      self.fee_rate,
-      script_pubkey,
-      burn_amount,
-    )?;
+        let burn_amount = Amount::from_sat(1);
+
+        (
+          Self::create_unsigned_burn_satpoint_transaction(
+            &wallet,
+            inscription_info.satpoint,
+            self.fee_rate,
+            script_pubkey,
+            burn_amount,
+          )?,
+          burn_amount,
+        )
+      }
+      Outgoing::Rune { decimal, rune } => {
+        ensure!(
+          self.cbor_metadata.is_none() && self.json_metadata.is_none(),
+          "metadata not supported for burning runes"
+        );
+
+        (
+          Self::create_unsigned_burn_runes_transaction(&wallet, rune, decimal, self.fee_rate)?,
+          Amount::ZERO,
+        )
+      }
+      _ => bail!("burning is only implemented for inscriptions and runes"),
```

Comment:
Let's make this exhaustive instead of a wildcard, and make the messages reflect that, i.e. "can't burn bitcoin", "can't burn satpoints", "can't burn sats", etc.

Reply:
We could easily support burning sats and satpoints, but we can do that later.